### DISCUSSION TITLE
Ionic ou React?

### DISCUSSION THREAD

**Discussion by romulo2735** [frontendbr/forum#1179] [Perguntas e Respostas]
Pessoal quero entrar no mundo mobile, e vejo que IONIC e REACT são as mais badaladas do mercado, antes disso estava aprendendo PHONE GAP + CORDOVA, e agora estou meio perdido, Qual metodo de mobile hibrido seguir? Ionic? React? os dois ou só um deles? Valeu.

**Comment by kivervinicius** [comment]
React Native, ionic utiliza webview (cordova) e é bem mais lento.

**Comment by fakenickels** [comment]
A performance do React Native é muito maior e escala melhor

**Comment by diegodsgarcia** [comment]
Como qualquer coisa de TI, tudo vai depender do projeto!
Não existe framework ou linguagem bala de prata.
React e Ionic tem suas vantagens e desvantagens. Caso sua aplicação não utilizar muitos recursos nativos do celular, e precisa fazer um projeto em um curto tempo, utilize Ionic.
Caso sua aplicação for muito complexa, utilizando recursos nativos (e tiver tempo e disponibilidade para criar telas e features) ai sim vá para um React da vida.

Como dizem, você não utiliza uma bazuca pra matar uma formiga...

**Comment by felipefialho** [comment]
Acho que além de performance, depende até mesmo das suas skills ou da equipe, Ionic usa Angular + TypeScript por debaixo dos panos, se tem mais facilidade com isso vai em frente...

**Comment by jeftarmascarenhas** [comment]
Simples, se você está apenas querendo estudar e já tem alguma experiência com angular o entendimento do Ionic é mais rápido, se tem conhecimento do React o entendimento do React Native é mais rápido.

Minha opinião: Se você tem a intenção de estudar pensando no mercado de trabalho, freelas... O React Nativo neste momento seria a melhor escolha na minha opinião.

**Comment by fakenickels** [comment]
> Caso sua aplicação não utilizar muitos recursos nativos do celular, e precisa fazer um projeto em um curto tempo, utilize Ionic.
Não concordo com essa afirmação.

Acho o custo de começar um projeto com React Native muito baixo, mais baixo até do que lidar com Ionic. Pode estar enviesado pq uso React todo dia, mas depois que usei RN pela primeira vez depois de sofrer muito com Ionic a diferença de experiência pra melhor foi imediata.

**Comment by romulo2735** [comment]
Valeu Pessoal!

**Comment by rickslayer** [comment]
Estou com a mesma dúvida e acabei achando um artigo (em inglês) com explicações bem detalhadas das duas ferramentas.
https://www.codementor.io/fmcorz/react-native-vs-ionic-du1087rsw

**Reply by paulo-social** [reply]
O Ionic mudou bastante desde 2016, inclusive hoje eles usam React e também tem recursos nativos.

**Comment by paulo-social** [comment]
O Ionic usa React hoje em dia, além de Angular e Vue. E também utiliza o capacitor com recursos nativos.
